### PR TITLE
Update cwltool pin to 3.1.20211107152837

### PR DESCRIPTION
**Pull request by nsoranzo**
## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/dependencies/dev-requirements.txt [hunk 1]
```diff
@@ -193,8 +193,7 @@ pytz==2021.3; python_version >= "3.6" and python_full_version < "3.0.0" or pytho
 pyuwsgi==2.0.19.1.post0
 pyyaml==6.0; python_version >= "3.6"
 pyzmq==22.3.0; python_version >= "3.6"
-rdflib-jsonld==0.5.0; python_version >= "3.6" and python_version < "4"
-rdflib==4.2.2; python_version >= "3.6" and python_version < "4"
+rdflib==5.0.0; python_version >= "3.6" and python_version < "4"
```
Comment (mr-c) [inline lib/galaxy/dependencies/dev-requirements.txt:196]:
```suggestion
rdflib==5.0.0
```
Since Python3.6 is your minimum, yes?

Comment (nsoranzo) [reply lib/galaxy/dependencies/dev-requirements.txt:196]: That's stuff added by poetry, it's not productive to fight it :smile:

Comment (mr-c) [reply lib/galaxy/dependencies/dev-requirements.txt:196]: Got it! I didn't know which was you and which was autogenerated

Comment (nsoranzo) [reply lib/galaxy/dependencies/dev-requirements.txt:196]: `pyproject.toml` is what we change manually (new requirements, pins...).

**Review by mr-c — APPROVED** [review]

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**